sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainerResponse.java

```diff
@@ -18,7 +21,11 @@
             super.setProperties(null);
             container = null;
         } else {
-            CosmosContainerProperties props = new CosmosContainerProperties(bodyAsString);
+            Callable<CosmosContainerProperties> createCosmosContainerPropertiesCallable = () -> {
+                return new CosmosContainerProperties(bodyAsString);
+            };
+            SerializationDiagnosticsContext serializationDiagnosticsContext = BridgeInternal.getSerializationDiagnosticsContext(this.getCosmosResponseDiagnostics());
+            CosmosContainerProperties props = serializationDiagnosticsContext.getResource(createCosmosContainerPropertiesCallable, SerializationDiagnosticsContext.SerializationType.ContainerSerialization);
```

Comment:
why are we preferring this approach over having a start timer and end timer before and after serialization?

this way you have to change every `deserialization()` code to a `() -> deserialization()` ?

Reply:
This way we need not to put start,end time and adding it to the serialization list , that logic can be confined in one place.  There are some places where we can't use getResource() due to unavailability of serialization context  such as in getCreateDocumentRequest() in RxDocumentClientImpl.java , so for those we have addSerializationDiagnostics() as you suggesting.

Reply:
removed callable as it was have some perf cost